UserGuide.md

```diff
@@ -181,9 +186,15 @@ It is encouraged to map outputs from your test WDL to inputs for your evaluation
 
 ##### <a name="specifying-custom-docker-images"/> **Specifying Custom Docker Images**
 
-CARROT can generate a docker image to supply as an input to your test WDL or evaluation WDL.  It must be possible to generate the docker image by pulling from a GitHub repo and submitting it to [Google Cloud Build](https://cloud.google.com/build/docs/overview).  In order to use a custom docker image, first a **[software](#carrot-terms-and-structure)** record must be created in CARROT, either using the REST API or carrot_cli, that references the GitHub repository.  Once that is done, the custom build can be specified as an input to either your test WDL or your evaluation WDL.  For example, if you have created an entry for a software called “<code><em>my_cool_software</em></code>” and you want to generate a docker image from the commit with the hash “<code><em>ca82a6dff817ec66f44342007202690a93763949</em></code>” to supply to an input named “<code><em>docker_image</em></code>” in a workflow titled “<code><em>Test_Workflow</em></code>” in your test WDL, your test input json should contain the following:
+CARROT can generate a docker image to supply as an input to your test WDL or evaluation WDL. It must be possible to generate the docker image by pulling from a GitHub repo and submitting it to [Google Cloud Build](https://cloud.google.com/build/docs/overview). The simplest way to enable this is to ensure that it is possible to build the docker image by running `docker build` from the root of the repo. If this is not possible, Google Cloud Build provides options for more complex builds.
+
+In order to use a custom docker image, first a **[software](#carrot-terms-and-structure)** record must be created in CARROT, either using the REST API or carrot_cli, that references the git repository.  Once that is done, the custom build can be specified as an input to either your test WDL or your evaluation WDL.  For example, if you have created an entry for a software called “<code><em>my_cool_software</em></code>” and you want to generate a docker image from the commit with the hash “<code><em>ca82a6dff817ec66f44342007202690a93763949</em></code>” to supply to an input named “<code><em>docker_image</em></code>” in a workflow titled “<code><em>Test_Workflow</em></code>” in your test WDL, your test input json should contain the following:
+
+`"Test_Workflow.docker_image":"image_build:my_cool_software|ca82a6dff817ec66f44342007202690a93763949"`
+
+It is also possible to specify the commit to build from using tags.  For example, if “<code><em>my_cool_software</em></code>” has a commit tagged “<code><em>0.5.9</em></code>”, you can build from that commit like this:
 
-`"Test_Workflow.docker_image":"image_build:my_cool_software|ca82a6dff817ec66f44342007202690a93763949`”
+`"Test_Workflow.docker_image":"image_build:my_cool_software|0.5.9"`
```

Comment:
What about branch names?  Can it take anything that is a valid git reference  (HEAD^) or is it only tags/hashes?